### PR TITLE
1905: Don't ignore dependencies in tasks inserted before start pt.

### LINKED ISSUE #1905: Refine pre-start ignore
Follows #1540 

Pre-start ignore currently wipes out *all* dependencies prior to the start point.  It should only needs to wipe out ones the cross the start point (it *may* be OK to ignore all prior intra-cycle dependence as a approximation)

### PR DESCRIPTION
**Pull request by arjclark**
Closes #1905 

This change means that any task inserted prior to the start point will maintain its prerequisites rather than having them wiped out. Existing behaviour maintained (no changes needed to existing tests) and new test added for this desired behaviour.

@hjoliver - please review 1
@benfitzpatrick - please review 2

### CODE REVIEW

File: lib/cylc/prerequisite.py [hunk 1]
```diff
@@ -109,7 +110,8 @@ def set_condition(self, expr):
                     try:
                         foo = m.group().split(".")[1].rstrip()
                         if get_point(foo) < self.start_point:
```
Comment (benfitzpatrick) [inline lib/cylc/prerequisite.py]: I guess we can combine the `if`s?

**Comment by benfitzpatrick** [comment]
In your example suite, if I insert the family at `T06` in held state, then insert it again at `T12`, I guess we don't get `bar[-PT6H] => foo`?

**Comment by arjclark** [comment]
> In your example suite, if I insert the family at T06 in held state, then insert it again at T12, I guess we don't get bar[-PT6H] => foo?

Not sure I understand your question - if the cycle at which a task is inserted is prior to the start cycle then its prerequisites are not filtered. You see this in the example suite as the task (foo) is set to ready in order for it to be able to run.

**Comment by arjclark** [comment]
@benfitzpatrick - if statement tweaked.

**Comment by arjclark** [comment]
@benfitzpatrick - I've added the extra test to ensure future continuation of functionality as we discussed yesterday.

**Comment by benfitzpatrick** [comment]
Ok, looks good to me

**Comment by benfitzpatrick** [comment]
Oops - merged without @hjoliver.

**Comment by benfitzpatrick** [comment]
Any retrospective approval?

**Comment by hjoliver** [comment]
Retrospectively tested and approved.  (However, an email coming on a related issue...)

**Pull request merged**